In [143]:
REPEAT_MASKER = "/home/analytics/anaconda3/envs/bioinf/bin/RepeatMasker"
BEDTOOLS = "/home/analytics/anaconda3/bin/bedtools"
UPDATE_BLASTDB = "/home/analytics/anaconda3/bin/update_blastdb.pl"
MAKEBLASTDB = "/home/analytics/anaconda3/bin/makeblastdb"
BLASTP = "/home/analytics/anaconda3/bin/blastp"

In [2]:
!mkdir -p ../data/week4

In [3]:
!wget -P ../data/week4/ ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/001/949/185/GCA_001949185.1_Rvar_4.0/GCA_001949185.1_Rvar_4.0_genomic.fna.gz

--2018-10-18 15:08:36--  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/001/949/185/GCA_001949185.1_Rvar_4.0/GCA_001949185.1_Rvar_4.0_genomic.fna.gz
           => ‘../data/week4/GCA_001949185.1_Rvar_4.0_genomic.fna.gz’
Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 130.14.250.13, 2607:f220:41e:250::13
Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|130.14.250.13|:21... connected.
Logging in as anonymous ... Logged in!
==> SYST ... done.    ==> PWD ... done.
==> TYPE I ... done.  ==> CWD (1) /genomes/all/GCA/001/949/185/GCA_001949185.1_Rvar_4.0 ... done.
==> SIZE GCA_001949185.1_Rvar_4.0_genomic.fna.gz ... 17621408
==> PASV ... done.    ==> RETR GCA_001949185.1_Rvar_4.0_genomic.fna.gz ... done.
Length: 17621408 (17M) (unauthoritative)

100%[======================================>] 17,621,408  4.87MB/s   in 3.9s   

2018-10-18 15:08:42 (4.30 MB/s) - ‘../data/week4/GCA_001949185.1_Rvar_4.0_genomic.fna.gz’ saved [17621408]



In [4]:
!gunzip ../data/week4/GCA_001949185.1_Rvar_4.0_genomic.fna.gz

In [6]:
!/home/analytics/anaconda3/envs/bioinf/bin/RepeatMasker -species elegans -s ../data/week4/GCA_001949185.1_Rvar_4.0_genomic.fna

RepeatMasker version open-4.0.7
Search Engine: NCBI/RMBLAST [ 2.6.0+ ]
Rebuilding RepeatMaskerLib.embl library
  - Read in 216 sequences from /home/analytics/anaconda3/envs/bioinf/share/RepeatMasker/Libraries/DfamConsensus.embl
RepeatMaskerLib.embl: 216 total sequences.
Master RepeatMasker Database: /home/analytics/anaconda3/envs/bioinf/share/RepeatMasker/Libraries/RepeatMaskerLib.embl ( Complete Database: dc20170127 )


Warning...unknown stuff <
>
Building general libraries in: /home/analytics/anaconda3/envs/bioinf/share/RepeatMasker/Libraries/dc20170127/general
Building species libraries in: /home/analytics/anaconda3/envs/bioinf/share/RepeatMasker/Libraries/dc20170127/caenorhabditis
   - 174 ancestral and ubiquitous sequence(s) for caenorhabditis
   - 0 lineage specific sequence(s) for caenorhabditis

analyzing file ../data/week4/GCA_001949185.1_Rvar_4.0_genomic.fna

Checking for E. coli insertion elements
identifying Simple Repeats in batch 1 of 976
identifying matches to caenorhabd

identifying Simple Repeats in batch 37 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 38 of 976
identifying matches to caenorhabditis sequences in batch 38 of 976
identifying Simple Repeats in batch 38 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 39 of 976
identifying matches to caenorhabditis sequences in batch 39 of 976
identifying Simple Repeats in batch 39 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 40 of 976
identifying matches to caenorhabditis sequences in batch 40 of 976
identifying Simple Repeats in batch 40 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 41 of 976
identifying matches to caenorhabditis sequences in batch 41 of 976
identifying Simple Repeats in batch 41 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 42 of 976
identifying matches to caenorhabditis sequences in batch 42 of 976


identifying Simple Repeats in batch 78 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 79 of 976
identifying matches to caenorhabditis sequences in batch 79 of 976
identifying Simple Repeats in batch 79 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 80 of 976
identifying matches to caenorhabditis sequences in batch 80 of 976
identifying Simple Repeats in batch 80 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 81 of 976
identifying matches to caenorhabditis sequences in batch 81 of 976
identifying Simple Repeats in batch 81 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 82 of 976
identifying matches to caenorhabditis sequences in batch 82 of 976
identifying Simple Repeats in batch 82 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 83 of 976
identifying matches to caenorhabditis sequences in batch 83 of 976


identifying Simple Repeats in batch 119 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 120 of 976
identifying matches to caenorhabditis sequences in batch 120 of 976
identifying Simple Repeats in batch 120 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 121 of 976
identifying matches to caenorhabditis sequences in batch 121 of 976
identifying Simple Repeats in batch 121 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 122 of 976
identifying matches to caenorhabditis sequences in batch 122 of 976
identifying Simple Repeats in batch 122 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 123 of 976
identifying matches to caenorhabditis sequences in batch 123 of 976
identifying Simple Repeats in batch 123 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 124 of 976
identifying matches to caenorhabditis sequences in ba

identifying Simple Repeats in batch 160 of 976
identifying matches to caenorhabditis sequences in batch 160 of 976
identifying Simple Repeats in batch 160 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 161 of 976
identifying matches to caenorhabditis sequences in batch 161 of 976
identifying Simple Repeats in batch 161 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 162 of 976
identifying matches to caenorhabditis sequences in batch 162 of 976
identifying Simple Repeats in batch 162 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 163 of 976
identifying matches to caenorhabditis sequences in batch 163 of 976
identifying Simple Repeats in batch 163 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 164 of 976
identifying matches to caenorhabditis sequences in batch 164 of 976
identifying Simple Repeats in batch 164 of 976

Checking for E. coli inse

identifying Simple Repeats in batch 200 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 201 of 976
identifying matches to caenorhabditis sequences in batch 201 of 976
identifying Simple Repeats in batch 201 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 202 of 976
identifying matches to caenorhabditis sequences in batch 202 of 976
identifying Simple Repeats in batch 202 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 203 of 976
identifying matches to caenorhabditis sequences in batch 203 of 976
identifying Simple Repeats in batch 203 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 204 of 976
identifying matches to caenorhabditis sequences in batch 204 of 976
identifying Simple Repeats in batch 204 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 205 of 976
identifying matches to caenorhabditis sequences in ba

identifying Simple Repeats in batch 241 of 976
identifying matches to caenorhabditis sequences in batch 241 of 976
identifying Simple Repeats in batch 241 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 242 of 976
identifying matches to caenorhabditis sequences in batch 242 of 976
identifying Simple Repeats in batch 242 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 243 of 976
identifying matches to caenorhabditis sequences in batch 243 of 976
identifying Simple Repeats in batch 243 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 244 of 976
identifying matches to caenorhabditis sequences in batch 244 of 976
identifying Simple Repeats in batch 244 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 245 of 976
identifying matches to caenorhabditis sequences in batch 245 of 976
identifying Simple Repeats in batch 245 of 976

Checking for E. coli inse

identifying Simple Repeats in batch 281 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 282 of 976
identifying matches to caenorhabditis sequences in batch 282 of 976
identifying Simple Repeats in batch 282 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 283 of 976
identifying matches to caenorhabditis sequences in batch 283 of 976
identifying Simple Repeats in batch 283 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 284 of 976
identifying matches to caenorhabditis sequences in batch 284 of 976
identifying Simple Repeats in batch 284 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 285 of 976
identifying matches to caenorhabditis sequences in batch 285 of 976
identifying Simple Repeats in batch 285 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 286 of 976
identifying matches to caenorhabditis sequences in ba

identifying Simple Repeats in batch 322 of 976
identifying matches to caenorhabditis sequences in batch 322 of 976
identifying Simple Repeats in batch 322 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 323 of 976
identifying matches to caenorhabditis sequences in batch 323 of 976
identifying Simple Repeats in batch 323 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 324 of 976
identifying matches to caenorhabditis sequences in batch 324 of 976
identifying Simple Repeats in batch 324 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 325 of 976
identifying matches to caenorhabditis sequences in batch 325 of 976
identifying Simple Repeats in batch 325 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 326 of 976
identifying matches to caenorhabditis sequences in batch 326 of 976
identifying Simple Repeats in batch 326 of 976

Checking for E. coli inse

identifying Simple Repeats in batch 362 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 363 of 976
identifying matches to caenorhabditis sequences in batch 363 of 976
identifying Simple Repeats in batch 363 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 364 of 976
identifying matches to caenorhabditis sequences in batch 364 of 976
identifying Simple Repeats in batch 364 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 365 of 976
identifying matches to caenorhabditis sequences in batch 365 of 976
identifying Simple Repeats in batch 365 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 366 of 976
identifying matches to caenorhabditis sequences in batch 366 of 976
identifying Simple Repeats in batch 366 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 367 of 976
identifying matches to caenorhabditis sequences in ba

identifying Simple Repeats in batch 403 of 976
identifying matches to caenorhabditis sequences in batch 403 of 976
identifying Simple Repeats in batch 403 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 404 of 976
identifying matches to caenorhabditis sequences in batch 404 of 976
identifying Simple Repeats in batch 404 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 405 of 976
identifying matches to caenorhabditis sequences in batch 405 of 976
identifying Simple Repeats in batch 405 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 406 of 976
identifying matches to caenorhabditis sequences in batch 406 of 976
identifying Simple Repeats in batch 406 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 407 of 976
identifying matches to caenorhabditis sequences in batch 407 of 976
identifying Simple Repeats in batch 407 of 976

Checking for E. coli inse

identifying Simple Repeats in batch 443 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 444 of 976
identifying matches to caenorhabditis sequences in batch 444 of 976
identifying Simple Repeats in batch 444 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 445 of 976
identifying matches to caenorhabditis sequences in batch 445 of 976
identifying Simple Repeats in batch 445 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 446 of 976
identifying matches to caenorhabditis sequences in batch 446 of 976
identifying Simple Repeats in batch 446 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 447 of 976
identifying matches to caenorhabditis sequences in batch 447 of 976
identifying Simple Repeats in batch 447 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 448 of 976
identifying matches to caenorhabditis sequences in ba

identifying Simple Repeats in batch 484 of 976
identifying matches to caenorhabditis sequences in batch 484 of 976
identifying Simple Repeats in batch 484 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 485 of 976
identifying matches to caenorhabditis sequences in batch 485 of 976
identifying Simple Repeats in batch 485 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 486 of 976
identifying matches to caenorhabditis sequences in batch 486 of 976
identifying Simple Repeats in batch 486 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 487 of 976
identifying matches to caenorhabditis sequences in batch 487 of 976
identifying Simple Repeats in batch 487 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 488 of 976
identifying matches to caenorhabditis sequences in batch 488 of 976
identifying Simple Repeats in batch 488 of 976

Checking for E. coli inse

identifying Simple Repeats in batch 524 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 525 of 976
identifying matches to caenorhabditis sequences in batch 525 of 976
identifying Simple Repeats in batch 525 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 526 of 976
identifying matches to caenorhabditis sequences in batch 526 of 976
identifying Simple Repeats in batch 526 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 527 of 976
identifying matches to caenorhabditis sequences in batch 527 of 976
identifying Simple Repeats in batch 527 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 528 of 976
identifying matches to caenorhabditis sequences in batch 528 of 976
identifying Simple Repeats in batch 528 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 529 of 976
identifying matches to caenorhabditis sequences in ba

identifying Simple Repeats in batch 565 of 976
identifying matches to caenorhabditis sequences in batch 565 of 976
identifying Simple Repeats in batch 565 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 566 of 976
identifying matches to caenorhabditis sequences in batch 566 of 976
identifying Simple Repeats in batch 566 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 585 of 976
identifying matches to caenorhabditis sequences in batch 585 of 976
identifying Simple Repeats in batch 585 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 586 of 976
identifying matches to caenorhabditis sequences in batch 586 of 976
identifying Simple Repeats in batch 586 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 587 of 976
identifying matches to caenorhabditis sequences in batch 587 of 976
identifying Simple Repeats in batch 587 of 976

Checking for E. coli inse

identifying Simple Repeats in batch 623 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 624 of 976
identifying matches to caenorhabditis sequences in batch 624 of 976
identifying Simple Repeats in batch 624 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 625 of 976
identifying matches to caenorhabditis sequences in batch 625 of 976
identifying Simple Repeats in batch 625 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 626 of 976
identifying matches to caenorhabditis sequences in batch 626 of 976
identifying Simple Repeats in batch 626 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 627 of 976
identifying matches to caenorhabditis sequences in batch 627 of 976
identifying Simple Repeats in batch 627 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 628 of 976
identifying matches to caenorhabditis sequences in ba

identifying Simple Repeats in batch 664 of 976
identifying matches to caenorhabditis sequences in batch 664 of 976
identifying Simple Repeats in batch 664 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 665 of 976
identifying matches to caenorhabditis sequences in batch 665 of 976
identifying Simple Repeats in batch 665 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 666 of 976
identifying matches to caenorhabditis sequences in batch 666 of 976
identifying Simple Repeats in batch 666 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 667 of 976
identifying matches to caenorhabditis sequences in batch 667 of 976
identifying Simple Repeats in batch 667 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 668 of 976
identifying matches to caenorhabditis sequences in batch 668 of 976
identifying Simple Repeats in batch 668 of 976

Checking for E. coli inse

identifying Simple Repeats in batch 704 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 705 of 976
identifying matches to caenorhabditis sequences in batch 705 of 976
identifying Simple Repeats in batch 705 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 706 of 976
identifying matches to caenorhabditis sequences in batch 706 of 976
identifying Simple Repeats in batch 706 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 707 of 976
identifying matches to caenorhabditis sequences in batch 707 of 976
identifying Simple Repeats in batch 707 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 708 of 976
identifying matches to caenorhabditis sequences in batch 708 of 976
identifying Simple Repeats in batch 708 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 709 of 976
identifying matches to caenorhabditis sequences in ba

identifying Simple Repeats in batch 745 of 976
identifying matches to caenorhabditis sequences in batch 745 of 976
identifying Simple Repeats in batch 745 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 746 of 976
identifying matches to caenorhabditis sequences in batch 746 of 976
identifying Simple Repeats in batch 746 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 747 of 976
identifying matches to caenorhabditis sequences in batch 747 of 976
identifying Simple Repeats in batch 747 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 748 of 976
identifying matches to caenorhabditis sequences in batch 748 of 976
identifying Simple Repeats in batch 748 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 749 of 976
identifying matches to caenorhabditis sequences in batch 749 of 976
identifying Simple Repeats in batch 749 of 976

Checking for E. coli inse

identifying Simple Repeats in batch 785 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 786 of 976
identifying matches to caenorhabditis sequences in batch 786 of 976
identifying Simple Repeats in batch 786 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 787 of 976
identifying matches to caenorhabditis sequences in batch 787 of 976
identifying Simple Repeats in batch 787 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 788 of 976
identifying matches to caenorhabditis sequences in batch 788 of 976
identifying Simple Repeats in batch 788 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 789 of 976
identifying matches to caenorhabditis sequences in batch 789 of 976
identifying Simple Repeats in batch 789 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 790 of 976
identifying matches to caenorhabditis sequences in ba

identifying Simple Repeats in batch 826 of 976
identifying matches to caenorhabditis sequences in batch 826 of 976
identifying Simple Repeats in batch 826 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 827 of 976
identifying matches to caenorhabditis sequences in batch 827 of 976
identifying Simple Repeats in batch 827 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 828 of 976
identifying matches to caenorhabditis sequences in batch 828 of 976
identifying Simple Repeats in batch 828 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 829 of 976
identifying matches to caenorhabditis sequences in batch 829 of 976
identifying Simple Repeats in batch 829 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 830 of 976
identifying matches to caenorhabditis sequences in batch 830 of 976
identifying Simple Repeats in batch 830 of 976

Checking for E. coli inse

identifying Simple Repeats in batch 866 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 867 of 976
identifying matches to caenorhabditis sequences in batch 867 of 976
identifying Simple Repeats in batch 867 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 868 of 976
identifying matches to caenorhabditis sequences in batch 868 of 976
identifying Simple Repeats in batch 868 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 869 of 976
identifying matches to caenorhabditis sequences in batch 869 of 976
identifying Simple Repeats in batch 869 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 870 of 976
identifying matches to caenorhabditis sequences in batch 870 of 976
identifying Simple Repeats in batch 870 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 871 of 976
identifying matches to caenorhabditis sequences in ba

identifying Simple Repeats in batch 907 of 976
identifying matches to caenorhabditis sequences in batch 907 of 976
identifying Simple Repeats in batch 907 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 908 of 976
identifying matches to caenorhabditis sequences in batch 908 of 976
identifying Simple Repeats in batch 908 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 909 of 976
identifying matches to caenorhabditis sequences in batch 909 of 976
identifying Simple Repeats in batch 909 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 910 of 976
identifying matches to caenorhabditis sequences in batch 910 of 976
identifying Simple Repeats in batch 910 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 911 of 976
identifying matches to caenorhabditis sequences in batch 911 of 976
identifying Simple Repeats in batch 911 of 976

Checking for E. coli inse

identifying Simple Repeats in batch 947 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 948 of 976
identifying matches to caenorhabditis sequences in batch 948 of 976
identifying Simple Repeats in batch 948 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 949 of 976
identifying matches to caenorhabditis sequences in batch 949 of 976
identifying Simple Repeats in batch 949 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 950 of 976
identifying matches to caenorhabditis sequences in batch 950 of 976
identifying Simple Repeats in batch 950 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 951 of 976
identifying matches to caenorhabditis sequences in batch 951 of 976
identifying Simple Repeats in batch 951 of 976

Checking for E. coli insertion elements
identifying Simple Repeats in batch 952 of 976
identifying matches to caenorhabditis sequences in ba

In [7]:
!wget -P ../data/week4/ http://public.dobzhanskycenter.ru/mrayko/BIMM185/augustus.whole.gff

--2018-10-18 17:28:41--  http://public.dobzhanskycenter.ru/mrayko/BIMM185/augustus.whole.gff
Resolving public.dobzhanskycenter.ru (public.dobzhanskycenter.ru)... 195.70.204.3
Connecting to public.dobzhanskycenter.ru (public.dobzhanskycenter.ru)|195.70.204.3|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 27539173 (26M)
Saving to: ‘../data/week4/augustus.whole.gff’

100%[======================================>] 27,539,173  69.2MB/s   in 0.4s   

2018-10-18 17:28:42 (69.2 MB/s) - ‘../data/week4/augustus.whole.gff’ saved [27539173/27539173]



In [11]:
!$BEDTOOLS getfasta -bed ../data/week4/augustus.whole.gff -fo ../data/week4/augustus.whole.fa


Tool:    bedtools getfasta (aka fastaFromBed)
Version: v2.27.1
Summary: Extract DNA sequences from a fasta file based on feature coordinates.

Usage:   bedtools getfasta [OPTIONS] -fi <fasta> -bed <bed/gff/vcf>

Options: 
	-fi	Input FASTA file
	-fo	Output file (opt., default is STDOUT
	-bed	BED/GFF/VCF file of ranges to extract from -fi
	-name	Use the name field for the FASTA header
	-name+	Use the name field and coordinates for the FASTA header
	-split	given BED12 fmt., extract and concatenate the sequences
		from the BED "blocks" (e.g., exons)
	-tab	Write output in TAB delimited format.
		- Default is FASTA format.

	-s	Force strandedness. If the feature occupies the antisense,
		strand, the sequence will be reverse complemented.
		- By default, strand information is ignored.

	-fullHeader	Use full fasta header.
		- By default, only the word before the first space or tab 
		is used.



In [12]:
!perl ../scripts/getAnnoFasta.pl ../data/week4/augustus.whole.gff

In [14]:
!ls ../data/week4/

augustus.whole.aa
augustus.whole.gff
GCA_001949185.1_Rvar_4.0_genomic.fna
GCA_001949185.1_Rvar_4.0_genomic.fna.cat.gz
GCA_001949185.1_Rvar_4.0_genomic.fna.masked
GCA_001949185.1_Rvar_4.0_genomic.fna.out
GCA_001949185.1_Rvar_4.0_genomic.fna.tbl


In [15]:
!cat ../data/week4/augustus.whole.aa | grep ">" | wc -l

16435


In [16]:
!wget -P ../data/week4/ http://public.dobzhanskycenter.ru/mrayko/BIMM185/peptides.fa

--2018-10-18 17:38:30--  http://public.dobzhanskycenter.ru/mrayko/BIMM185/peptides.fa
Resolving public.dobzhanskycenter.ru (public.dobzhanskycenter.ru)... 195.70.204.3
Connecting to public.dobzhanskycenter.ru (public.dobzhanskycenter.ru)|195.70.204.3|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1340 (1.3K)
Saving to: ‘../data/week4/peptides.fa’

100%[======================================>] 1,340       --.-K/s   in 0s      

2018-10-18 17:38:30 (78.0 MB/s) - ‘../data/week4/peptides.fa’ saved [1340/1340]



In [17]:
!head -n 10 ../data/week4/peptides.fa

>pept_1_score_48
VFSALDVLR
>pept_2_score_65
SPLQTDEIR
>pept_3_score_54
AATGAVLSSADSR
>pept_4_score_39
LLDDENDYELK
>pept_5_score_59
VTGSSQGAINQQQAK


In [18]:
!cat ../data/week4/peptides.fa | wc -l

86


In [121]:
!head ../data/week4/augustus.whole.aa

>g1.t1
AATYSKHPRSYRENIVVWRICMGHCDGRYLALYNLNGWSERDMIDRWALMVTLLVPPAFLLPARDIDNLLTAGHETSPQTGGSDVMSEYVTPFCYVCDSL
VDADCRESQDWMPRRRNGLRKKVMWTGQGEDDHGRLAFDWKEVYIRLEKFKQRCPSNNGCTTHVFLNSDRVIRRCFDPKEPPCSGHEWMVNRRNPKKRSK
VKVVSRLTCADWPFCNVDSAYYWPNGTVEALLSPHDDPQLKVRSAKLSCYECDSILQPDCTGQNFVKAVRKPNDAVAHSVDWVGLRKGMEKFRTRCPVES
GCFFQGDHMTNRVIRGCRNMTTDLPRHLLVKDRRKESSVFRTEEVCRESECNSRVLFFSHSGAFLDSDWHFNNSSAGRDNGQGEAKKEVYQGDAPATEET
GFLPHVKQSGQGKRKEGRLQKRTIGDVEREERRADEARQRRGKPKAEGRGGAGTRVLSFGHRSRKTSAGGKLDSCWLLLLLPFFLL
>g2.t1
MTFAVTVATTSQSCAIAFDVFWLTTIKLFFIAKYFGSEFTPFDKVQSLTCPHCAQQGIEYKDLLPHLETNHADTSDFLVVSDNITIIWDVAHCI
>g3.t1
MLDKAKLVLLNKGGRPSIRSAEDGWKVFSDSLLCRASSDGRVFIGKGGGNGGAVAKAEVAAIGVVLVHHMLQQVVRHGEVVIVSANTNWTPLPY


In [21]:
!$MAKEBLASTDB -in ../data/week4/augustus.whole.aa -dbtype prot -out ../data/week4/proteindb   



Building a new DB, current time: 10/18/2018 17:41:33
New DB name:   /home/analytics/projects/bioinf_workshop/data/week4/proteindb
New DB title:  ../data/week4/augustus.whole.aa
Sequence type: Protein
Keep MBits: T
Maximum file size: 1000000000B
Adding sequences from FASTA; added 16435 sequences in 0.675385 seconds.


In [23]:
!$BLASTP -query ../data/week4/peptides.fa -db ../data/week4/proteindb > ../data/week4/blast_result

In [119]:
!cat ../data/week4/blast_result | grep -e "^>" | awk '{ print $2 }' | sort | uniq > ../data/week4/dna_att_genes

In [122]:
proteins = {}
cur_seq = []

with open("../data/week4/augustus.whole.aa", "r") as f:
    for line in f:
        if line[0] == ">":
            if len(cur_seq) > 0:
                proteins[cur_prot] = "".join(cur_seq)
                cur_seq = []
            cur_prot = line[1:].strip()
        else:
            cur_seq.append(line.strip())

In [123]:
with open("../data/week4/to_psort", "w") as f:
    dap = !cat ../data/week4/dna_att_genes
    for p in dap:
        f.write(">" + p + "\n")
        f.write(proteins[p] + "\n")

In [125]:
!head -n 4 ../data/week4/to_psort

>g10513.t1
MSTTTSSSSSNKDKDSTDTYVRSADNTSGSSNTTGSSSTNKGGKSSSSDYDSDKTTTSSSYGTGGQNTSSYGQSGQGGQHNTSTSSSYGQSGQGGQSGQHSSSSYGQSGQHSSGQSGQHSSGQSGQQSGQHGSHEVQQKLKEVGNLLQKAGHLLQDLQGSASDFSSSSSYSQRNQGGNYSGPYGGSDSQFHQYGSSGMGGGSYGQSSSYGQGSSGYGQGSSSYGQQSGRHQDSDRFGGSSSFGGQSSGQYGGHSQGGYGGQQGGYGGSSGQNYGPYXXXXXXXXXXXXXRGMGGGYGFSQDSGRQGGMGMSGGMGGADRYGGFGGPNRPDMSYGQQYGGRSDRW
>g10514.t1
MSYNRTEYRSDSDRHDDDKQGGWRSWFGLGGNKDDNNDRDRGYSYTNTTTYRGDDSNRYGFSGDRMSGPSGYSGAISGGYSGSRGGYGYTTSDDYNRGNTNYSRSNDNSNYNRDNSSRGGDRDVYYRQETRTYPTTGYTGSSNYTSGXXXXXXXXXXXXXXXXXXXXXXXXXXPSYGNTDRDYTVSXXXXXXXXXXXXXXXXXXXXXGYGFSGDRDTSFRNTSFNTTGDRDTFNRGSYSGGYATSGNQGGFSTTSGMGHPGNYSSSYTSPSGTYGQSSNYSYNRNY


The response from WOLFPSOT

In [127]:
wolf_response = "g10513.t1 details nucl: 20, cyto_nucl: 14.5, cyto: 7, extr: 3, E.R.: 1, golg: 1 g10514.t1 details nucl: 19, cyto_nucl: 15, cyto: 9, extr: 3, mito: 1 g11320.t1 details plas: 24.5, extr_plas: 16, extr: 6.5, lyso: 1 g11513.t1 details cyto: 17, cyto_nucl: 12.8333, cyto_mito: 9.83333, nucl: 7.5, E.R.: 3, mito: 1.5, plas: 1, pero: 1, golg: 1 g11806.t1 details nucl: 18, cyto_nucl: 11.8333, mito: 5, extr: 4, cyto: 3.5, cyto_pero: 2.66667, cysk_plas: 1 g11960.t1 details nucl: 32 g12388.t1 details extr: 25, plas: 4, mito: 2, lyso: 1 g12510.t1 details plas: 29, cyto: 3 g12562.t1 details extr: 30, lyso: 2 g1285.t1 details extr: 25, plas: 5, mito: 1, lyso: 1 g13530.t1 details extr: 13, nucl: 6.5, lyso: 5, cyto_nucl: 4.5, plas: 3, E.R.: 3, cyto: 1.5 g14472.t1 details nucl: 28, plas: 2, cyto: 1, cysk: 1 g15153.t1 details extr: 32 g15484.t1 details nucl: 17.5, cyto_nucl: 15.3333, cyto: 12, cyto_mito: 6.83333, plas: 1, golg: 1 g16318.t1 details nucl: 20.5, cyto_nucl: 13, extr: 5, cyto: 4.5, E.R.: 1, golg: 1 g16368.t1 details nucl: 20.5, cyto_nucl: 13, extr: 5, cyto: 4.5, E.R.: 1, golg: 1 g2203.t1 details plas: 29, nucl: 2, golg: 1 g3428.t1 details mito: 18, cyto: 11, extr: 2, nucl: 1 g3679.t1 details extr: 26, mito: 2, lyso: 2, plas: 1, E.R.: 1 g4106.t1 details E.R.: 14.5, E.R._golg: 9.5, extr: 7, golg: 3.5, lyso: 3, pero: 2, plas: 1, mito: 1 g4970.t1 details plas: 32 g5237.t1 details plas: 24, mito: 8 g5443.t1 details extr: 28, nucl: 3, cyto: 1 g5467.t1 details extr: 27, plas: 4, mito: 1 g5502.t1 details extr: 31, lyso: 1 g5503.t1 details extr: 29, plas: 1, mito: 1, lyso: 1 g5510.t1 details plas: 23, mito: 7, E.R.: 1, golg: 1 g5616.t1 details extr: 31, mito: 1 g5641.t1 details extr: 31, lyso: 1 g5927.t1 details nucl: 30.5, cyto_nucl: 16.5, cyto: 1.5 g702.t1 details extr: 29, plas: 2, lyso: 1 g7861.t1 details nucl: 16, cyto_nucl: 14, cyto: 8, plas: 5, pero: 1, cysk: 1, golg: 1 g8100.t1 details nucl: 16.5, cyto_nucl: 12.5, cyto: 7.5, plas: 5, extr: 2, E.R.: 1 g8312.t1 details nucl: 15.5, cyto_nucl: 15.5, cyto: 12.5, mito: 2, plas: 1, golg: 1"

In [54]:
import re

In [128]:
cur_span = 0
strings = []

for match in re.finditer(r"g\d+\.t\d+", wolf_response):
    next_span = match.span()[0]
    strings.append(wolf_response[cur_span:next_span].strip())
    cur_span = next_span
    
strings = strings[1:]

In [130]:
predictions = dict()

for s in strings:
    ss = s.split(" ")
    predictions[ss[0]] = ss[2][:-1]

In [131]:
predictions

{'g10513.t1': 'nucl',
 'g10514.t1': 'nucl',
 'g11320.t1': 'plas',
 'g11513.t1': 'cyto',
 'g11806.t1': 'nucl',
 'g11960.t1': 'nucl',
 'g12388.t1': 'extr',
 'g12510.t1': 'plas',
 'g12562.t1': 'extr',
 'g1285.t1': 'extr',
 'g13530.t1': 'extr',
 'g14472.t1': 'nucl',
 'g15153.t1': 'extr',
 'g15484.t1': 'nucl',
 'g16318.t1': 'nucl',
 'g16368.t1': 'nucl',
 'g2203.t1': 'plas',
 'g3428.t1': 'mito',
 'g3679.t1': 'extr',
 'g4106.t1': 'E.R.',
 'g4970.t1': 'plas',
 'g5237.t1': 'plas',
 'g5443.t1': 'extr',
 'g5467.t1': 'extr',
 'g5502.t1': 'extr',
 'g5503.t1': 'extr',
 'g5510.t1': 'plas',
 'g5616.t1': 'extr',
 'g5641.t1': 'extr',
 'g5927.t1': 'nucl',
 'g702.t1': 'extr',
 'g7861.t1': 'nucl',
 'g8100.t1': 'nucl'}

In [132]:
targetp_output = """g10513.t1             344          0.199  0.026  0.870   _    2
g10514.t1             286          0.088  0.044  0.933   _    1
g11320.t1             466          0.014  0.962  0.068   S    1
g11513.t1            1099          0.079  0.095  0.900   _    1
g11806.t1             201          0.509  0.074  0.448   M    5
g11960.t1             981          0.488  0.043  0.512   _    5
g12388.t1             165          0.547  0.721  0.015   S    5
g12510.t1             513          0.124  0.243  0.753   _    3
g12562.t1             200          0.037  0.873  0.184   S    2
g1285.t1              193          0.290  0.846  0.028   S    3
g13530.t1             931          0.038  0.918  0.147   S    2
g14472.t1             445          0.113  0.049  0.910   _    2
g15153.t1             177          0.051  0.845  0.157   S    2
g15484.t1             903          0.095  0.037  0.934   _    1
g16318.t1             264          0.085  0.052  0.933   _    1
g16368.t1             244          0.081  0.054  0.935   _    1
g2203.t1              845          0.107  0.176  0.810   _    2
g3428.t1              171          0.352  0.047  0.706   _    4
g3679.t1              285          0.136  0.862  0.023   S    2
g4106.t1              642          0.018  0.982  0.039   S    1
g4970.t1              901          0.063  0.063  0.939   _    1
g5237.t1              228          0.696  0.067  0.309   M    4
g5443.t1              120          0.039  0.872  0.122   S    2
g5467.t1              145          0.394  0.898  0.012   S    3
g5502.t1              244          0.266  0.879  0.021   S    2
g5503.t1              215          0.267  0.879  0.021   S    2
g5510.t1              208          0.273  0.137  0.604   _    4
g5616.t1              202          0.032  0.895  0.170   S    2
g5641.t1              192          0.056  0.861  0.122   S    2
g5927.t1              874          0.153  0.042  0.901   _    2
g702.t1               215          0.054  0.876  0.141   S    2
g7861.t1              619          0.291  0.005  0.900   _    2
g8100.t1             1313          0.130  0.081  0.856   _    2
g8312.t1              886          0.051  0.075  0.940   _    1"""

In [133]:
targetp_lines = [l.split() for l in targetp_output.split("\n")][2:]

In [134]:
targetp_predicted = [l for l in targetp_lines if l[5] == "_"]

In [135]:
wolf_set = set([k for k, v in predictions.items() if v =="nucl"])
targetp_set = set([l[0] for l in targetp_predicted])

In [138]:
with open("../data/week4/to_blast", "w") as f:
    for g in wolf_set.intersection(targetp_set):
        f.write(">" + g + "\n")
        f.write(proteins[g] + "\n")

In [136]:
print("\n\n".join([proteins[s] for s in wolf_set.intersection(targetp_set)]))

PYGGFDSQSQPYGGFDSQSQPYGGFDSQSQPYGGFDSQSQPYGGFDSQSQPYGGFDSQSQPYGGFDSQSQPYGGFDSQSQPYGGSDSQFQLYEPSGMGGGSYGQSSSYGQGSSGYGQGSSSYGQQSGRPQDSDRFGGSSSFDGQPSGQYGGHSQGGYGGQQGGYGGSNGQSYGPYGSSSNQQSSMESNRGMDGSYGFSQDSGRQGGMGMPDFVGGSDRYGGFEGPNRPDMSYGQQNGGRIGDRW

MASTHQSSTEPSSTGKSEETKKDASQGSGQDSKNVTVTKGTGSSATSAAIVKTGGSQGKDSSTTAGSSSTQGQKFSTTPTDPKTFSSDQKEKSKSPAKEVPSGGDSKSQGDTKSQSDAKSSGQSQGQSKDSGKSSSDSSKSHSVIGAVKDVVAGAKDVAGKAVEDAPSIMHTAVDAVKNAATTVKDVASSAASTVAEKVVDAYHSVVGDKTDDKKEGEHSGDKKDDSKAGSGSGQGGDNKKSEGETSGQAESSSGNEGAAPAKGRGRGRPPAAAKGVAKGAAKGAAASKGAKSGAESSKGGEQSSGDIEMADASSKGGSDQRDSAATVGEGGASGSEGGAKKGRGRGAGKKADAGDTSAEPPRRSSRLTSSGTGAGSAPAAAKGGAKRAASSSSTPSNAKKQATGGAGKAAATKATAAKSAASKAPQNGAGAKKKGGKAGGRKRK

MEAMKISAKTHEESANSSHNGSSSSLKMENGMTHGIYAKMAKEKLNASHLAERINSTERLKKESTEHLSHLEDGPRRSTSEDAELFDTAILKDLEFAKTLNERGEISISPATSEDSLVLRALRRSDYHLEYMELLSQLTVVGAVTQDMFEKRFDSMRRAEGSYYTIVIEEKLPKRRIVGSATLVIEQKFIRATALHVRIEELLGRMWTVSTHYRLMEVATVLSKSLGCYKVSLESMSTEADSQVEPASAPTPAVKSTTSQPAPQSTPSVTANTPSSPQLSPNPTPSHSQEIAVGHTPDKSKSHSTAN

In [144]:
!$UPDATE_BLASTDB swissprot

Connected to NCBI


In [145]:
!mkdir -p ../data/week4/swissprot
!mv ./swissprot* ../data/week4/swissprot
!tar -xf ../data/week4/swissprot/swissprot.tar.gz
# !MAKEBLASTDB -in ../data/week4/swissprot/ -dbtype prot -out ../data/week4/proteindb   

In [148]:
!$BLASTP -query ../data/week4/to_blast -db ../data/week4/swissprot/swissprot > ../data/week4/swissprot_blast_result

In [151]:
!head -n 50 ../data/week4/swissprot_blast_result

BLASTP 2.7.1+


Reference: Stephen F. Altschul, Thomas L. Madden, Alejandro A.
Schaffer, Jinghui Zhang, Zheng Zhang, Webb Miller, and David J.
Lipman (1997), "Gapped BLAST and PSI-BLAST: a new generation of
protein database search programs", Nucleic Acids Res. 25:3389-3402.


Reference for composition-based statistics: Alejandro A. Schaffer,
L. Aravind, Thomas L. Madden, Sergei Shavirin, John L. Spouge, Yuri
I. Wolf, Eugene V. Koonin, and Stephen F. Altschul (2001),
"Improving the accuracy of PSI-BLAST protein database searches with
composition-based statistics and other refinements", Nucleic Acids
Res. 29:2994-3005.



Database: Non-redundant UniProtKB/SwissProt sequences
           470,430 sequences; 177,485,237 total letters



Query= g16368.t1

Length=244
                                                                      Score     E
Sequences producing significant alignments:                          (Bits)  Value

A4II09.1  RecName: Full=Eukaryotic translation initiation factor

hypothetical protein RvY_12593 [Ramazzottius varieornatus]	580	580	100%	0.0	96%	GAV01971.1
RecName: Full=Damage suppressor protein	814	814	100%	0.0	100%	P0DOW4.1
Select seq GAV10013.1	hypothetical protein RvY_19500 [Ramazzottius varieornatus]	394	394	100%	4e-137	97%	GAV10013.1
hypothetical protein RvY_09739 [Ramazzottius varieornatus]	728	728	27%	0.0	100%	GAU98615.1
hypothetical protein RvY_19422 [Ramazzottius varieornatus]	426	426	100%	3e-149	97%	GAV09945.1
hypothetical protein RvY_07107 [Ramazzottius varieornatus]	1224	1224	73%	0.0	96%	GAU95500.1
Select seq GAV03924.1	hypothetical protein RvY_14288 [Ramazzottius varieornatus]	2008	2008	100%	0.0	100%	GAV03924.1
Select seq GAU98932.1	hypothetical protein RvY_10004 [Ramazzottius varieornatus]	1731	1731	94%	0.0	99%	GAU98932.1
Select seq GAV08752.1	hypothetical protein RvY_18400 [Ramazzottius varieornatus]	1482	1482	78%	0.0	100%	GAV08752.1
Select seq GAU98300.1	hypothetical protein RvY_09464 [Ramazzottius varieornatus]	1236	1236	100%	0.0	97%	GAU98300.1